app/models/caution_flag.rb

```diff
@@ -2,6 +2,9 @@
 
 class CautionFlag < ApplicationRecord
   belongs_to :institution, counter_cache: :count_of_caution_flags
+  scope :unique_flag, lambda {
```

Comment:
I have a couple thoughts about the name for this scope.  I think it should be plural because it returns more than one in the relation.  Secondly I think it should have the word 'distinct' in it because using a `DISTINCT` in a query affects other things you could do to the relation, like call `count`.  So my recommendation would be to call this scope `distinct_flags`

Reply:
changed